### ISSUE TITLE
0.35.0: segmentation fault when trying to use --vo=gpu-next

### ISSUE THREAD

**Issue by tgurr** [mpv-player/mpv#10891]
### Important Information

Provide following Information:
- mpv version
```
mpv 0.35.0 Copyright © 2000-2022 mpv/MPlayer/mplayer2 projects
 built on Fri Nov 18 00:00:44 CET 2022
FFmpeg library versions:
   libavutil       57.28.100
   libavcodec      59.37.100
   libavformat     59.27.100
   libswscale      6.7.100
   libavfilter     8.44.100
   libswresample   4.7.100
FFmpeg version: 5.1.2
```
- Linux Distribution and Version
`Exherbo Linux`
- Source of the mpv binary
`Compiled on local system (Exherbo is a source-based distribution)`
- If known which version of mpv introduced the problem
`unknown`
- Window Manager and version
`kwin_x11 5.26.3`
- GPU driver and version
`proprietary NVIDIA 520.56.06 including proprietary kernel driver`
- Possible screenshot or video of visual glitches
`none`

### Reproduction steps
Reproducible with any video when using `--no-config --vo=gpu-next`, e.g.:

1. `yt-dlp -S ext:mp4:m4a https://www.youtube.com/watch?v=F7IZZXQ89Oc`
2. `mpv --no-config --vo=gpu-next --log-file=test.log Turn\ it\ up\ to\ 11\ \[F7IZZXQ89Oc\].mp4`

### Expected behavior
Be able to play any video file using `--vo=gpu-next`.

### Actual behavior
```
$ mpv --no-config --vo=gpu-next --log-file=test.log Turn\ it\ up\ to\ 11\ \[F7IZZXQ89Oc\].mp4 
 (+) Video --vid=1 (*) (h264 320x182 14.985fps)
 (+) Audio --aid=1 (*) (aac 2ch 44100Hz)
Segmentation fault (core dumped)
```

### Log file
[test.log](https://github.com/mpv-player/mpv/files/10042540/test.log)
[coredump.txt](https://github.com/mpv-player/mpv/files/10042561/coredump.txt)

### Sample files
`yt-dlp -S ext:mp4:m4a https://www.youtube.com/watch?v=F7IZZXQ89Oc`

### Additional information
Things work fine when not using `--vo=gpu-next`, my regular mpv.conf:
```
hwdec=auto
cache=no
gpu-api=vulkan
ao=pipewire
```
Please let me know if/how I can contribute further information helping to identify the issue.

**Comment by haasn** [comment]
Seems related to https://github.com/mpv-player/mpv/pull/10856

Some sort of bug is causing `opts` to remain uninitialized, it seems. I'm curious why it only seems to happen on your environments, since I cannot reproduce this myself.

Does it crash consistently for you or only very rarely?

**Comment by tgurr** [comment]
> Seems related to #10856

Indeed! Thanks for pointing out that PR, with the patch applied things started to work for me as well.

**Comment by tgurr** [comment]
> Does it crash consistently for you or only very rarely?

Sorry didn't see your question before, it doesn't work at all for me, like the segfault is reproducible 100% trying to play any video file. GCC version used to compile is 12.2.0. Not sure if it may be related but since there's some sort of mentioning of icc, I've a icc profile in use for colord monitor calibration, but switching it back to the default icc doesn't seem to make any difference either. Maybe NVIDIA driver related as the author of the patch also mentioned being on proprietary NVIDIA driver?

**Comment by haasn** [comment]
If you apply this patch, does it also crash on `--vo=gpu`?

```diff
diff --git a/video/out/vo_gpu.c b/video/out/vo_gpu.c
index 404b7e77d7..d0108a61cb 100644
--- a/video/out/vo_gpu.c
+++ b/video/out/vo_gpu.c
@@ -291,6 +291,7 @@ static int preinit(struct vo *vo)
 
     struct ra_ctx_opts *ctx_opts = mp_get_config_group(vo, vo->global, &ra_ctx_conf);
     struct gl_video_opts *gl_opts = mp_get_config_group(vo, vo->global, &gl_video_conf);
+    assert(gl_opts->icc_opts->size_str);
     struct ra_ctx_opts opts = *ctx_opts;
     opts.want_alpha = gl_opts->alpha_mode == 1;
     p->ctx = ra_ctx_create(vo, opts);
```

**Comment by tgurr** [comment]
> If you apply this patch, does it also crash on `--vo=gpu`?

Yes, with that patch `--vo=gpu` also fails, on:
`mpv: ../video/out/vo_gpu.c:296: preinit: Assertion 'gl_opts->icc_opts->use_embedded' failed.`
